Makefile

```diff
@@ -60,33 +60,43 @@ SPECTEST_VERSION := $(shell cat .spectest_version)
 SPECTEST_CONFIGS = general minimal mainnet
 
 SPECTEST_ROOTDIR = test/spec/vectors
+SPECTEST_GENERATED_ROOTDIR = test/generated
+VECTORS_DIR = $(SPECTEST_ROOTDIR)/tests
 # create directory if it doesn't exist
 $(info $(shell mkdir -p $(SPECTEST_ROOTDIR)))
 
 SPECTEST_DIRS := $(patsubst %,$(SPECTEST_ROOTDIR)/tests/%,$(SPECTEST_CONFIGS))
+SPECTEST_GENERATED := $(patsubst %,$(SPECTEST_GENERATED_ROOTDIR)/%,$(SPECTEST_CONFIGS))
 SPECTEST_TARS := $(patsubst %,$(SPECTEST_ROOTDIR)/%_${SPECTEST_VERSION}.tar.gz,$(SPECTEST_CONFIGS))
 
 $(SPECTEST_ROOTDIR)/%_${SPECTEST_VERSION}.tar.gz:
 	curl -L -o "$@" \
 		"https://github.com/ethereum/consensus-spec-tests/releases/download/${SPECTEST_VERSION}/$*.tar.gz"
 
-$(SPECTEST_ROOTDIR)/tests/%: $(SPECTEST_ROOTDIR)/%_${SPECTEST_VERSION}.tar.gz
+$(VECTORS_DIR)/%: $(SPECTEST_ROOTDIR)/%_${SPECTEST_VERSION}.tar.gz
 	-rm -rf $@
 	tar -xzmf "$<" -C $(SPECTEST_ROOTDIR)
 
+$(SPECTEST_GENERATED_ROOTDIR): $(VECTORS_DIR)
+	mix generate_spec_tests
+
 download-vectors: $(SPECTEST_TARS)
 
 clean-vectors:
 	-rm -rf $(SPECTEST_ROOTDIR)/tests
 	-rm $(SPECTEST_ROOTDIR)/*.tar.gz
 
+clean-tests:
+	-rm -r test/generated
+
+gen-spec:
+	mix generate_spec_tests
```

Comment:
Shouldn't this be `gen-spec: $(SPECTEST_GENERATED)`?

Reply:
Fixed. Depending on the generated dir.